offline/coturn.md

```diff
@@ -0,0 +1,471 @@
+# Installing Coturn.
+
+Coturn is a free and open-source implementation of TURN and STUN server. 
+
+It is used to relay media between two clients that are unable to establish a direct connection. 
+
+This is useful in cases where the clients are behind a NAT or a firewall.
+
+This document explains how to install Coturn on a newly deployed Wire-Server installation.
+
+This presumes you already have:
+
+* Followed the [single Hetzner machine installation](single_hetzner_machine_installation.md) guide or otherwise have a machine ready to accept a Wire-Server deployment.
+* Have followed the [Wire-Server installation](docs_ubuntu_22.04.md) guide and have Wire-Server deployed and working (with Restund as the TURN server, which is currently the default, and will be replaced by Coturn as part of this process).
+
+## Plan. 
+
+To setup Coturn, we will:
+
+* Create a `values.yaml` file and fill it with configuration.
+* Create a `secret.yaml` file for the Coturn secrets.
+* Configure the Coturn labels to select on which machine(s) it will run.
+* Configure the SFT labels for Coturn and SFT to share a port range.
+* Configure the port redirection in Nftables.
+* Change the Wire-Server configuration to use Coturn instead of Restund.
+* Disable Restund.
+* Install Coturn using Helm.
+* Verify that Coturn is working.
+
+This entire document presumes you are working from inside your Wire-Server deployment directory (typically `~/wire-server-deploy/`).
+
+Step by step:
+
+## Create a `values.yaml` file and fill it with configuration.
+
+Create a folder for the Coturn configuration:
+
+```bash
+
+mkdir -p values/coturn
+
+```
+
+Create/edit a `values.yaml` file inside the `values/coturn` folder:
+
+```bash
+
+nano values/coturn/values.yaml
+
+```
+
+Add the following configuration to the `values.yaml` file:
+
+```yaml
+
+# Value file for coturn chart.
+#
+# See: https://github.com/wireapp/wire-server/blob/develop/charts/coturn/values.yaml
+# And: https://github.com/wireapp/wire-server/blob/develop/charts/coturn/README.md
+
+nodeSelector:
+  wire.com/role: coturn
+
+coturnTurnListenIP: '192.168.122.23'
```

Comment:
shouldn't we have a listen IP, and a publish IP?

Reply:
annotate the node with the outside IP

Reply:
same for coturn as it is for sft

Reply:
fixed in latest commit.